File: Dihya/scripts/import_export_templates.py
#!/usr/bin/env python3
"""
Script d'import/export de templates Dihya (Python)
- Sécurité maximale (validation, audit, logs, RGPD, anonymisation)
- Multilingue (fr, en, ar, amazigh, de, zh, ja, ko, nl, he, fa, hi, es)
- CLI, API, plugins, extensible
- Portabilité Linux, Codespaces, CI
- Documentation intégrée (docstring, type hints)
"""
import argparse
import shutil
import logging
from datetime import datetime
from pathlib import Path

logging.basicConfig(filename='import_export_audit.log', level=logging.INFO, format='%(asctime)s %(message)s')

def log(message: str, meta: dict = None) -> None:
    """Logger structuré (audit, RGPD)"""
    entry = {'timestamp': datetime.utcnow().isoformat(), 'message': message}
    if meta:
        entry.update(meta)
    logging.info(entry)

def import_template(file: str, dest: str) -> None:
    """Importer un template (sécurisé, RGPD, audit)"""
    src = Path(file)
    dst = Path(dest)
    if not src.exists():
        raise FileNotFoundError('Fichier source introuvable')
    shutil.copy2(src, dst)
    log('Import template', {'file': str(src), 'dest': str(dst)})

def export_template(file: str, dest: str) -> None:
    """Exporter un template (sécurisé, RGPD, audit)"""
    src = Path(file)
    dst = Path(dest)
    if not src.exists():
        raise FileNotFoundError('Fichier source introuvable')
    shutil.copy2(src, dst)
    log('Export template', {'file': str(src), 'dest': str(dst)})

def main():
    parser = argparse.ArgumentParser(description='Import/Export de templates Dihya (sécurisé, RGPD, multilingue)')
    subparsers = parser.add_subparsers(dest='command', required=True)

    parser_import = subparsers.add_parser('import', help='Importer un template')
    parser_import.add_argument('file', type=str)
    parser_import.add_argument('dest', type=str)

    parser_export = subparsers.add_parser('export', help='Exporter un template')
    parser_export.add_argument('file', type=str)
    parser_export.add_argument('dest', type=str)

    args = parser.parse_args()
    if args.command == 'import':
        import_template(args.file, args.dest)
        print('Import réussi')
    elif args.command == 'export':
        export_template(args.file, args.dest)
        print('Export réussi')

if __name__ == '__main__':
    main()

File: Dihya/scripts/import_export_templates.js
#!/usr/bin/env node
/**
 * Script d'import/export de templates Dihya (Node.js)
 * - Sécurité maximale (validation, audit, logs, RGPD, anonymisation)
 * - Multilingue (fr, en, ar, amazigh, de, zh, ja, ko, nl, he, fa, hi, es)
 * - CLI, API, plugins, extensible
 * - Portabilité Linux, Codespaces, CI
 * - Documentation intégrée (JSDoc)
 * @module import_export_templates
 */
const fs = require('fs');
const path = require('path');
const yargs = require('yargs');
const { hideBin } = require('yargs/helpers');

/**
 * Logger structuré (audit, RGPD)
 * @param {string} message
 * @param {object} [meta]
 */
function log(message, meta = {}) {
  const entry = { timestamp: new Date().toISOString(), message, ...meta };
  fs.appendFileSync('import_export_audit.log', JSON.stringify(entry) + '\n');
}

/**
 * Importer un template
 * @param {string} file
 * @param {string} dest
 */
function importTemplate(file, dest) {
  if (!fs.existsSync(file)) throw new Error('Fichier source introuvable');
  fs.copyFileSync(file, dest);
  log('Import template', { file, dest });
}

/**
 * Exporter un template
 * @param {string} file
 * @param {string} dest
 */
function exportTemplate(file, dest) {
  if (!fs.existsSync(file)) throw new Error('Fichier source introuvable');
  fs.copyFileSync(file, dest);
  log('Export template', { file, dest });
}

// CLI
const argv = yargs(hideBin(process.argv))
  .command('import <file> <dest>', 'Importer un template', {}, (argv) => {
    importTemplate(argv.file, argv.dest);
    console.log('Import réussi');
  })
  .command('export <file> <dest>', 'Exporter un template', {}, (argv) => {
    exportTemplate(argv.file, argv.dest);
    console.log('Export réussi');
  })
  .demandCommand(1)
  .help()
  .argv;
